libraries/ArduinoOTA/ArduinoOTA.cpp

```diff
@@ -146,26 +146,26 @@ int ArduinoOTAClass::parseInt(){
   uint8_t index;
   char value;
   while(_udp_ota->peek() == ' ') _udp_ota->read();
-  for(index = 0; index < sizeof(data); ++index){
+  for(index = 0; index < sizeof(data); index++){
     value = _udp_ota->peek();
     if(value < '0' || value > '9'){
       data[index++] = '\0';
       return atoi(data);
     }
-    data[index++] = _udp_ota->read();
+    data[index] = _udp_ota->read();
   }
   return 0;
 }
 
 String ArduinoOTAClass::readStringUntil(char end){
   String res = "";
-  int value;
+  char value;
```

Comment:
What is the motivation for this change?

Reply:
value should be declared int, because read() returns an int that can be -1 on error

Reply:
The motivation is back to original before yoursunny PR (it was char) and homogenize data types ( '/0' and end are char)